### ISSUE TITLE
terraform destroy fails for data.local_file that defines a dependency to a null_resource in a module

### ISSUE THREAD

**Issue by camprod** [hashicorp/terraform#17983]
### Terraform Version
```
v0.11.7
```

### Terraform Configuration Files
_main.tf_
```
module "readjson" {
  source  = "./modules/readjson"
}

output "content" {
  value = "${module.readjson.content}"
}
```
_modules/readjson/readjson.tf_
```
resource "null_resource" "do_nothing" {
}

data "local_file" "samplejson" {
  filename  = "${path.module}/sample.json"
  depends_on = ["null_resource.do_nothing"]
}

output "content" {
  value = "${data.local_file.samplejson.content}"
}
```
_modules/readjson/sample.json_
```
{
  "name": "platform",
  "uid": "platform",
}
```
### Debug Output

[https://gist.github.com/camprod/0abcbd1ac4fccbf40413e66056940280](https://gist.github.com/camprod/0abcbd1ac4fccbf40413e66056940280)
### Expected Behavior

terraform destroy to succeed
### Actual Behavior

terraform destroy failed with 
```1 error(s) occurred:

* module.readjson.output.content: Resource 'data.local_file.samplejson' does not have attribute 'content' for variable 'data.local_file.samplejson.content'
```
### Steps to Reproduce

1. terraform init
2. terraform apply
3. terraform destroy

### Additional Context

If I remove `depends_on = ["null_resource.do_nothing"]` from readjson.tf , destroy works. Also, the following non-modularized version works: 

```
resource "null_resource" "do_nothing" {
}

data "local_file" "samplejson" {
  filename  = "${path.module}/sample.json"
  depends_on = ["null_resource.do_nothing"]
}

output "content" {
  value = "${data.local_file.samplejson.content}"
}
```

**Comment by jbardin** [comment]
Thanks for filing the issue with the complete example @camprod!

This will be fixed by #17942, and covered in #17976 and #17862

**Comment by jbardin** [comment]
Oh, and I should note that you do not want `depends_on` in a data source, as it will create a perpetual diff that can't be resolved.
